seatunnel-e2e/seatunnel-spark-connector-v2-e2e/src/test/java/org/apache/seatunnel/e2e/spark/v2/phoenix/FakeSourceToIoPhoenixIT.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.e2e.spark.v2.phoenix;
+
+import org.apache.seatunnel.e2e.spark.SparkContainer;
+
+import com.google.common.collect.Lists;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.Container;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.lifecycle.Startables;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.List;
+import java.util.stream.Stream;
+
+@Slf4j
+public class FakeSourceToIoPhoenixIT extends SparkContainer {
+
+    private static final String PHOENIX_DOCKER_IMAGE = "iteblog/hbase-phoenix-docker:1.0";
+    private static final String PHOENIX_HOST = "flink_e2e_phoenix_sink";
+    private static final int PHOENIX_PORT = 8765;
+    private static final String PHOENIX_CONNECT_URL = "jdbc:phoenix:thin:url=http://localhost:8765;serialization=PROTOBUF";
+    private static final String PHOENIX_JDBC_DRIVER = "org.apache.phoenix.queryserver.client.Driver";
+
+    private GenericContainer<?> phoenixServer;
+
+    @BeforeEach
+    public void startPhoenixContainer() throws Exception {
+        phoenixServer = new GenericContainer<>(PHOENIX_DOCKER_IMAGE)
+                .withNetwork(NETWORK)
+                .withNetworkAliases(PHOENIX_HOST)
+                .withLogConsumer(new Slf4jLogConsumer(log));
+        phoenixServer.setPortBindings(Lists.newArrayList(
+                String.format("%s:8765", PHOENIX_PORT)));
+        Startables.deepStart(Stream.of(phoenixServer)).join();
+        log.info("phoenix container started");
+        // wait for phoenix fully start
+        Thread.sleep(5000L);
```

Comment:
Use awaitility ?

case:
https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-e2e/seatunnel-flink-connector-v2-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/jdbc/JdbcGreenplumIT.java#L78

Reply:
I've fix it